app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/NightscoutUploader.java

```diff
@@ -710,12 +710,22 @@ private static synchronized void handleRestFailure(String msg) {
         Log.e(TAG, msg);
     }
 
+    private String getDeviceString(BgReading record) {
+        String withMethod = "xDrip-" + prefs.getString("dex_collection_method", "BluetoothWixel");
```

Comment:
Wondering about using `DexCollectionType.getBestCollectorHardwareName()` instead of the raw `dex_collection_method` preference item.

Reply:
I think that'd make sense... However because  "G5 Native" gets added to the source info already when that's in use as it is now this would lead to a duplicated "xDrip-G5 Native G5 Native" from the current "xDrip-DexcomG5 Native G5"/"xDrip-Dexcom G5 Native G5::Backfill"